src/app/hooks/useTasks.ts

```diff
@@ -1,57 +1,84 @@
-import { useState, useEffect } from "react";
 import { Task } from "../types/task";
+import useSWR, { mutate } from "swr";
+import { useState } from "react";
 
-export function useTasks(userId: string | undefined) {
-  const [tasks, setTasks] = useState<Task[]>([]);
-
-  useEffect(() => {
-    if (userId) {
-      fetchTasks();
-    }
-  }, [userId]);
+const fetcher = async (url: string) => {
+  const response = await fetch(url);
+  if (!response.ok) {
+    throw new Error("Failed to fetch tasks");
+  }
+  return response.json();
+};
 
-  const fetchTasks = async () => {
-    try {
-      const response = await fetch("/api/listTasks");
-      if (!response.ok) {
-        throw new Error("Failed to fetch tasks");
-      }
-      const data = await response.json();
-      setTasks(data);
-    } catch (error) {
-      console.error("Failed to fetch tasks:", error);
-    }
-  };
+export function useTasks(userId: string | undefined) {
+  const { data: tasks = [], error } = useSWR<Task[]>(
+    userId ? "/api/list-tasks" : null,
+    fetcher
+  );
+  const [isAdding, setIsAdding] = useState(false);
 
   const addTask = async (formData: FormData) => {
+    setIsAdding(true);
     try {
-      const response = await fetch("/api/createTask", {
+      const formDataTask = {
+        title: formData.get("title") as string,
```

Comment:
**Suggestion: Remove the as string cast and just use zod instead**

Reply:
done